### ISSUE TITLE
Test fails in daylight saving time TZs

### ISSUE THREAD

**Issue by akx** [python-babel/babel#370]
`test_get_timezone_name()`:

```
>       assert (dates.get_timezone_name(None, locale='en_US') ==                                      
                dates.get_timezone_name(datetime.now().replace(tzinfo=dates.LOCALTZ), locale='en_US'))
E       assert 'Eastern European Summer Time' == 'Eastern Europ...Standard Time'                      
E         - Eastern European Summer Time                                                              
E         ?                   ^^^^                                                                    
E         + Eastern European Standard Time                                                            
E         ?                   ^^^^^ +                                                                 
```

I think this is a LOCALTZ bug (since LOCALTZ is janky at best), but can't really be sure...

Note: should probably see if this is a regression introduced in 7800e547ed1710b24ee5679994cee6055c9c5698

**Comment by akx** [comment]
The same error also manifests on my OSX box.

**Comment by jtwang** [comment]
I see it too.

```
>       assert (dates.get_timezone_name(None, locale='en_US') ==
                dates.get_timezone_name(datetime.now().replace(tzinfo=dates.LOCALTZ), locale='en_US'))
E       assert 'Pacific Daylight Time' == 'Pacific Standard Time'
E         - Pacific Daylight Time
E         + Pacific Standard Time`
```

**Comment by BlackYoup** [comment]
Also happens on Linux:

```python
>       assert (dates.get_timezone_name(None, locale='en_US') ==
                dates.get_timezone_name(datetime.now().replace(tzinfo=dates.LOCALTZ), locale='en_US'))
E       assert 'Central European Summer Time' == 'Central Europ...Standard Time'
E         - Central European Summer Time
E         ?                   ^^^^
E         + Central European Standard Time
E         ?                   ^^^^^ +
```

**Comment by akx** [comment]
It's a regular timezone party!

**Comment by akx** [comment]
Thanks for the additional data point, @BlackYoup. :)

**Comment by akx** [comment]
Oh, right, this was actually closed by #411.
